engine/gamesys/src/gamesys/scripts/script_physics.cpp

```diff
@@ -1453,6 +1552,42 @@ namespace dmGameSystem
         return 0;
     }
 
+    static int Physics_SetListener(lua_State* L)
+    {
+        DM_LUA_STACK_CHECK(L, 0);
+        int top = lua_gettop(L);
+        int functionref = 0;
+        if (top >= 1) // completed cb
+        {
+            if (lua_isfunction(L, 1))
+            {
+                lua_pushvalue(L, 1);
+                // NOTE: By convention m_FunctionRef is offset by LUA_NOREF, in order to have 0 for "no function"
+                functionref = dmScript::RefInInstance(L) - LUA_NOREF;
```

Comment:
I must admit that I don't have a full grasp on when this goes out of scope.
Will it be destroyed when the collection is destroyed?
Or will it linger in the Lua space, since it has a positive refcount?
(I assume it will be destroyed, or we would have had similar issues before, right?)

Reply:
We have this issue https://github.com/defold/defold/issues/8123 , also this one https://github.com/defold/extension-spine/issues/147
not sure it's somehow related to what you are talking about.